fuzzy-native/src/MatcherBase.cpp

```diff
@@ -35,12 +35,51 @@ inline string str_to_lower(const std::string &s) {
   return lower;
 }
 
+float distance_to_root(const std::string *root, const std::string *value) {
+  size_t num_common_dirs = 0;
+  size_t i = 0;
+
+  // Count number of common directories
+  for (; i < root->length() && i < value->length(); i++) {
+    if (root->at(i) != value->at(i)) {
+      break;
+    }
+    if (root->at(i) == '/' || root->at(i) == '\\') {
+      num_common_dirs++;
+    }
+  }
+
+  if (i == root->length()) {
+    return 0;
+  }
+
+  // Count total number of directories in root
+  size_t num_dirs_root = num_common_dirs;
+  for (size_t j = i; j < root->length(); j++) {
+    if (root->at(j) == '/' || root->at(j) == '\\') {
+      num_dirs_root++;
+    }
+  }
+
+  float distance = num_dirs_root - num_common_dirs;
+  float penalty = 1;
+  for (size_t j = i; j < value->length(); j++) {
+    if (value->at(j) == '/' || value->at(j) == '\\') {
```

Comment:
We should precalculate the number of slashes in each candidate (see the `CandidateData`  struct) so we can avoid having to traverse the string again. Then this formula would be a function of `(num_candidate_dirs  - num_common_dirs)`.